Refactoring of airodump-ng and other related files

ChrisNisbet01 opened the pull request:
This is a fairly significant change, so I appreciate that you may be reluctant to take this without some serious thought and/or testing. I'm just bringing it to your attention. Do with it what you will.
Note that I had added some UBUS support and another output format for my own purposes, but removed all of that in this branch so that it more closely follows the support in your project.
The general idea from my perspective was to (hopefully) make the code a little clearer and modular.
Everything still builds on my local machine, and airodump-ng still appears to work, but I haven't checked any of the other applications to ensure they're still operational; I wouldn't know what they're used for anyway.
I haven't made any attempt to follow your own coding standards, but I figure that if you're interested in these changes, it wouldn't take too much effort to run a formatter over them.
I know that at least a couple of the new headers files are probably in the wrong place.

ChrisNisbet01 committed:
Continuing with work on getting rid of SIGUSR1 handler. Ultimate goal is to get teh mai process to read data from the child processes, not the signal handler.

ChrisNisbet01 committed:
The main process now reads the channel/frequency hopper process data rather than a signal handler.
WIP.

ChrisNisbet01 committed:
Removed some unnecessary memset.

ChrisNisbet01 committed:
Fixed build errors.

ChrisNisbet01 committed:
Started work on applying the Digi patch. Not sure what it's meant to be trying to do yet. Looks to be some other sort of output format.

ChrisNisbet01 committed:
Building again after applying the Digi patch code.

ChrisNisbet01 committed:
Added a dumy set_ht_channel callback into the file device so that airodump-ng will run when specifying a 'file://" device.

ChrisNisbet01 committed:
Formatting changes.

ChrisNisbet01 committed:
Signals (except for SIGSEGV) are now piped through to the main process for handling.

ChrisNisbet01 committed:
Did some cleanup.
Wi-Fi capture now uses a separate file pointer.

ChrisNisbet01 committed:
Did some cleanup.

ChrisNisbet01 committed:
Removed a pile of unnecessary memsets. Did some other cleanup.

ChrisNisbet01 committed:
Started work on replacing MAC addresses stored in char buffers with a typedef mac address and accompanying helper functions.
WIP.

ChrisNisbet01 committed:
Increased the maximum interface name length so that the app will work with long filenames as interfaces.

ChrisNisbet01 committed:
Ignore some Slickedit project files.

ChrisNisbet01 committed:
Refactored the list cleanup a little.

ChrisNisbet01 committed:
Changed char buffers used to store mac addresses so they now use the mac_address type and associated utilities.

ChrisNisbet01 committed:
Started looking into cleaning up lists.

ChrisNisbet01 committed:
NA list now uses generic TAILQ.

ChrisNisbet01 committed:
Started work on using a generic TAILQ list for AP entries. Still not building.
WIP.

ChrisNisbet01 committed:
APs are now entered into a generic TAILQ linked list.

ChrisNisbet01 committed:
Added lookup functions for the NA and AP lists to reduce code duplication.

ChrisNisbet01 committed:
Refactored getchancount() function.

ChrisNisbet01 committed:
Replaced custom linked lists in ST_info structs with generic TAILQ.

ChrisNisbet01 committed:
Packet lists now use a generic TAILQ (should change to singly linked list).
Made a change so that old packets are purged from the list. Previously this required packets of a certain type to keep arriving to purge old packets.

ChrisNisbet01 committed:
Now using a temporary TAILQ to store AP entries during sorting.

ChrisNisbet01 committed:
STAs now use a temporary list when sorting entries.

ChrisNisbet01 committed:
Started refactoring dump_print.
WIP.

ChrisNisbet01 committed:
Refactored creation of AP_info and ST_info structs.
Little bit of cleanup of the wifi_scanner output.

ChrisNisbet01 committed:
Minor cleanup/refactoring.

ChrisNisbet01 committed:
Moved the wifi scanner dump code into a separate module. Created a simple API to simplify the usage.

ChrisNisbet01 committed:
More refactoring. NA_info creation separated out.
Separated out cap file writing.

ChrisNisbet01 committed:
Started looking into sorting out the list mutexes, which hasn't been done correctly if I understand what was done correctly. Still not that happy with this.

ChrisNisbet01 committed:
Removed one call to dump_print, which seems to be unnecessary.

ChrisNisbet01 committed:
Moved all the code that reads packets from a file into a separate module.

ChrisNisbet01 committed:
Renamed packets_per_second function.

ChrisNisbet01 committed:
Factored out closing of output files.

ChrisNisbet01 committed:
Factored out the quit reset handling.

ChrisNisbet01 committed:
Cleaned up the console control a wee bit.

ChrisNisbet01 committed:
Refactored the sorting to remove an enormouse great switch stuffed in the middle of the sorting function.

ChrisNisbet01 committed:
Removed a couple of huge buffers that are no longer required.

ChrisNisbet01 committed:
Refactored some of the terminal control code.

ChrisNisbet01 committed:
replaced magic number -1 used to indicate invalid channel with an enum value.

ChrisNisbet01 committed:
Removed unnecessary setup_cards() function.

ChrisNisbet01 committed:
Made an interface name array a local variable as the caller wound up with an array of pointers to invalid memory, which would be bad if he didn't realise this.

ChrisNisbet01 committed:
Did a little bit of cleanup around the selected AP code.

ChrisNisbet01 committed:
Changed main() so that there is now a single exit point, and the terminal is restored as necessary.
Considered using an atexit() function, but wasn't sure how a fork() of the frequency/channel hopper process might interfere, so kept away from this.

ChrisNisbet01 committed:
Added support for freeing entries that haven't been seen for a period of time.
Specified using the '-v <minutes>' command line option.
Default is disabled.

ChrisNisbet01 committed:
Made the failed read counter interface specific.

ChrisNisbet01 committed:
Did some work on cleaning up the wifi scanner output.

ChrisNisbet01 committed:
Moved the CSV dump code into a separate module.

ChrisNisbet01 committed:
Changed the output dumping API. The application now receives an abstract type and doesn't call dump type specific methods directly.

ChrisNisbet01 committed:
dump_write Now using fprintf_mac_address() helper to print out MAC addresses to files.

ChrisNisbet01 committed:
Move some oui related code out of dump_write.
All oui code should eventually go into a separate module.

ChrisNisbet01 committed:
Kismet output files now use the new API.

ChrisNisbet01 committed:
kismet_netxml dump format now uses the new API.

ChrisNisbet01 committed:
Moved the OUI lookup related code into a separate module.
This code still needs cleanup.

ChrisNisbet01 committed:
Added an oui.txt file to aid with testing of the OUI code.

ChrisNisbet01 committed:
Moved battery related code into a separate module.

ChrisNisbet01 committed:
Removed unused 'keyout' field.

ChrisNisbet01 committed:
Cleaned up the gps related code a little.
Also removed some unused args from the original log initialiser code.

ChrisNisbet01 committed:
Moved the GPS tracker thread into a separate module.

ChrisNisbet01 committed:
Removed a call to dump_sort() from the main thread as I don't think it's required.

ChrisNisbet01 committed:
Factored out all the cleanup code from main into a separate function.

ChrisNisbet01 committed:
Moved setting of start time to just before it is used to create a start time string.

ChrisNisbet01 committed:
Moved the AP sorting code into a separate module.

ChrisNisbet01 committed:
Made the AP sort comparison struct an opaque type.

ChrisNisbet01 committed:
renamed ap_sort.[ch] ap_compare.[ch].

ChrisNisbet01 committed:
Completed rename operation.

ChrisNisbet01 committed:
Fixed an issue with the battery printout, which should only take place when doing GPS readings (I think).
Fixed a problem with the update interval, which was getting set to the incorrect (very large) value if the user input a -i arg <= 0.

ChrisNisbet01 committed:
Started refactoring channel/frequency hopping code.

ChrisNisbet01 committed:
Did some refactoring to the monitor mode/channel/frequency checking.

ChrisNisbet01 committed:
Renamed some timestamps

ChrisNisbet01 committed:
We bit of cleanup around failed interface reads. Got rid of a magic number.

ChrisNisbet01 committed:
Moved the code for the channel hopper process into a separate module.

ChrisNisbet01 committed:
Added dummy support for setting/getting the channel into the 'file' interface.
Useful for testing purposes.

ChrisNisbet01 committed:
Moved the code for the frequency hopper process into the channel hopper module.
Yes, it should probably go into a separate module, but they're so similar that it seemed reasonable to put them together (for now).

ChrisNisbet01 committed:
Factored out the packet capture code from the main loop.
made some changes to the way errors/timeouts are handled, which seemed overly complicated.

ChrisNisbet01 committed:
Added the 'nodecloak option to the help screen. Disabling this option prevents packets from being queued up, thereby saving memory.

ChrisNisbet01 committed:
Made some changes to the way the program terminates on error.

ChrisNisbet01 committed:
Changed a char array to a mac address type.

ChrisNisbet01 committed:
Fix some build warnings.

ChrisNisbet01 committed:
Made some changes to the includes to get things building under OpenWRT.
trying to work out openwrt build issues.

ChrisNisbet01 committed:
Added a couple of CFLAGS so that extra warnings show up.

ChrisNisbet01 committed:
Fixed some build warnings generated as a result of the compile time flags changes.

ChrisNisbet01 committed:
Renamed a local result variable and inverted its state.

ChrisNisbet01 committed:
Wee bit of cleanup in channel allocation code.

ChrisNisbet01 committed:
Temp debug comment out code that adds a packet to the list in an effort to track down a memory leak.

ChrisNisbet01 committed:
Reinstated the commented out code.

ChrisNisbet01 committed:
Did some formatting changes.

ChrisNisbet01 committed:
There appeared to be an issue with the sorting.

ChrisNisbet01 committed:
A resort is done if the sort order is inverted.

ChrisNisbet01 committed:
Did some refactoring of the wifi dump code.

Commit:
Added siome missing consts into the wifi scanner dump module.

Commit:
Renamed packet_read to pcap_reader.

Commit:
Updated pcap reader header names.

Commit:
renamed packet_reader.[ch] to pcap_reader.[ch]

Commit:
Changed the STA sorting to match the way AP sorting is done. This should get them displaying in the desried order on the console.

Commit:
Litle bit of cleanup.

Commit:
Refactored the bssid filtering.

Commit:
Whitepsace change.

ChrisNisbet01 committed:
Made a change so that APs are sorted at least every 5 seconds.

ChrisNisbet01 committed:
Fixing compile issues on OpenWRT.

ChrisNisbet01 committed:
Added some locking around access to the AP list.
Note that I think this whole business of having two threads manipulating the lists is broken. Better if a single thread does all the reading/writing of the list.

ChrisNisbet01 committed:
Moved all user input handling into the main thread. I wasn't at all keen on having multiple threads reading/writing the AP/STA lists.
I don't see that there's much to be gained by trying to offload the console output into a different thread, esp. in the wifi_scanner case where there is no user input.
Removal of locks to come.

ChrisNisbet01 committed:
Removed the locking around the various lists as it's no longer required. It didn't work correctly anyway.

ChrisNisbet01 committed:
Removed an unneeded ALLEGE() in gps_tracker.

ChrisNisbet01 committed:
Replaced an unaligned access with a load16.

ChrisNisbet01 committed:
Fiddling with AP sorting.

ChrisNisbet01 committed:
Added a couple of ubus-related files
Added some initial changes to add configure support for including UBUS support. Currently just adds a hashdef to config.h.

ChrisNisbet01 committed:
Started thinking about adding ubus support. Not sure if it's possible to include ubus and have separate file handling. I think the app is forced to do everythin via ubus/uloop files/timers, which would mean more reworking of the original code.

ChrisNisbet01 committed:
Refactored the oui code a little bit.

ChrisNisbet01 committed:
ignore build artifact

ChrisNisbet01 committed:
Made a change so that the unique IV tables are only allocated if configured to log IVs. Previously, memory would get allocated over time on the basis of received WEP packets whether or not the user was interested in these IVs.

ChrisNisbet01 committed:
Little bit of cleanup relating to the channel tables and node/packet purging.

ChrisNisbet01 committed:
More cleanup. Reducing dependency on global 'lopt' variable.

ChrisNisbet01 committed:
factored out the ap and na packets_per_second updates.

ChrisNisbet01 committed:
factored out the rx quality updates.
Moved output file/purging code into better locations.
Refactored quit request handling.

ChrisNisbet01 committed:
Moved the PCAP writing code out into a separate .[ch] module. Abstracted this out a bit as well.

ChrisNisbet01 committed:
Removed some references to the global lopt variable.

ChrisNisbet01 committed:
Removed an unused field from lopt.

ChrisNisbet01 committed:
Refactored the AP sorting. Grouped related variables into a structure.

ChrisNisbet01 committed:
User input handler no longer references the global lopt variable.

ChrisNisbet01 committed:
Removed more references to the global lopt variable.

ChrisNisbet01 committed:
Cleaned up the essid storage a bit.

ChrisNisbet01 committed:
Refactored AP and STA node lookup/creation.

ChrisNisbet01 committed:
Refactored dump output contexts to use an array rather than having separate fields/names.

ChrisNisbet01 committed:
airodump-ng no longer uses the 'opt' global variable, although it sttill needs to have one so that it will link with one of the libraries. Big sigh.

ChrisNisbet01 committed:
Removed some direct references to the global lopt variable.

ChrisNisbet01 committed:
Removed some direct references to the global lopt variable.

ChrisNisbet01 committed:
Removed access to global options variable lopt by the dump routines.

ChrisNisbet01 committed:
Moved the essid filter context init and clenaup into the essid_filter module.
Moved the quitting flags into the options structure as there is no need for them any more.

ChrisNisbet01 committed:
More cleanup.
Fixed a wee issue with pause control.

ChrisNisbet01 committed:
Refactored the pause control/display a little.

ChrisNisbet01 committed:
Added a 'utils' module and moved the time_as_string function into it.

ChrisNisbet01 committed:
Abstracted out the AP sort context mathods a little. Haven't made a 'proper' ADT as that would required more work.

ChrisNisbet01 committed:
options no longer passed to get_channels() as it only needs a pointer to an integer array to be allocated.

ChrisNisbet01 committed:
Refactored the AP sorting a little.

ChrisNisbet01 committed:
Changed the AP sort comparison a little from >= to > so that equal nodes don't get switch with each sort.

ChrisNisbet01 committed:
Refactored the AP filtering a little to reduce code duplication.

ChrisNisbet01 committed:
Removed the separate periodic sort required calculation. Periodic sorting is now done as part of the generic refresh.

ChrisNisbet01 committed:
Refactored the creation of AP and STA nodes a little.

ChrisNisbet01 committed:
Reducing dependency levels where appropriate.

ChrisNisbet01 committed:
Logically grouped some related fields in the options structure.

ChrisNisbet01 committed:
Cleaned up relative time handling a little.

ChrisNisbet01 committed:
Added support for _not_ including a prefix in the output filename, or looking for the first free index.

ChrisNisbet01 committed:
The user input thread no longer references the local options structure directly. It is passed all the data it needs by the the thread arg.

ChrisNisbet01 committed:
Removed some unneeded memsets.

ChrisNisbet01 committed:
Cleaned up the terminal output a little. Fixed a nasty macro so that it no longer has hidden side-effects.

ChrisNisbet01 committed:
Moved wait_proc into the utils module.

ChrisNisbet01 committed:
Renamed a update_console_output() to update_console().

ChrisNisbet01 committed:
Extracted out the code that reads a packet from an interface.

ChrisNisbet01 committed:
Duh. Console message wasn't getting printed after the user paused.

ChrisNisbet01 committed:
The terminal output is set back to normal/white at program termination.

ChrisNisbet01 committed:
Factored out the file packet reading code into a separate function.

ChrisNisbet01 committed:
Wee bit of cleanup.

ChrisNisbet01 committed:
Used check screen macro to replace some hard-coded equivalent code.

ChrisNisbet01 committed:
wifi_scanner output mode no longer puts the app into background mode.
Specify background (non-interactive) mode on the command line by using the -K 1 option.

ChrisNisbet01 committed:
Moved a call to gps_tracker initialise into the 'start' call. This function should always be called before starting, so let's not give the caller a chance not to.

ChrisNisbet01 committed:
Added a utility function to make ESSID values with control characters in them (i.e. < ' ') printable to reduce code duplication.

ChrisNisbet01 committed:
Removed some unused variables. Reduced the scope of some others.

ChrisNisbet01 committed:
Refactored removal of nodes from lists a little to reduce code duplication.
Also clean up allocation of NA nodes.

ChrisNisbet01 committed:
Moved a couple of utility functions into utils.h.

ChrisNisbet01 committed:
Started cleaning up dump_add_packet().

ChrisNisbet01 committed:
More cleaning up of dump_add_packet().

ChrisNisbet01 committed:
Added a command line option to disable shared key checks, which may result in an unexpected 'xor' output file getting created.

ChrisNisbet01 committed:
Replaced some magic numbers with defined values.

ChrisNisbet01 committed:
Reduced the scope of a mac address variable.

ChrisNisbet01 committed:
Renamed and moved an essid/iv logging related variable

ChrisNisbet01 committed:
Extracted out the ivs logging code from add_dump_packet.
This also reduces code duplication.

ChrisNisbet01 committed:
Moved the ivs logging code into a separate module.

ChrisNisbet01 committed:
Extracted out the key stream logging into the ivs_log module.

ChrisNisbet01 committed:
Moved the code that opens the ivs log file and writes the initial header into ivs_log.c

ChrisNisbet01 committed:
Moved the log_csv file open code into dump_write.

ChrisNisbet01 committed:
Moved parse_timestamp() into utils.

ChrisNisbet01 committed:
Extracted the AP update related code from dump_add_packet().

ChrisNisbet01 committed:
Moved the sta_update code into a separate function.

ChrisNisbet01 committed:
Fixed a bug in the oui initialisation. Duh.

ChrisNisbet01 committed:
Did some refactoring to do away with one of the gotos in dump_add_packet.
Reduced the scope of some other variables.

ChrisNisbet01 committed:
Added a utility function to search for control characters.

ChrisNisbet01 committed:
Removed some duplicated string data from the console message.

ChrisNisbet01 committed:
Created a utility function to peform moving average calc.

ChrisNisbet01 committed:
AP sorting wasn't working when sort order was inverted.

ChrisNisbet01 committed:
Two of the comparison functions were broken.

ChrisNisbet01 committed:
Made the AP sort context an ADT.
Contexts are now allocated/freed.
Removed the 'default' sort method.

ChrisNisbet01 committed:
Starting to reduce the size of dump_add_packet by extracting code into smaller functions.

ChrisNisbet01 committed:
Beacon and association packet parsing pulled out of dump_add_packet().

ChrisNisbet01 committed:
Packet data and control frame parsing pulled out of dump_add_packet().

ChrisNisbet01 committed:
Removed the last of the references to the global options variable.
main() and the signal handler are the only two functions to reference it.

ChrisNisbet01 committed:
Moved the code that sends probe requests into a separate module.

ChrisNisbet01 committed:
Moved some functions around to make the layout more consistent.

ChrisNisbet01 committed:
Renamed some options. Changed the state of the shared_key variable.

ChrisNisbet01 committed:
Moved some functions around so they're grouped more logically.

ChrisNisbet01 committed:
Extracted out the decloaking checks.

ChrisNisbet01 committed:
Changed a calloc() size from sizeof(int) to sizeof element.

ChrisNisbet01 committed:
Extracted out the generic update (5 second interval) into a separate function.
This may may 'ubusifying' the code easier later on.

ChrisNisbet01 committed:
Renamed the nodecloak option no-decloak so that it better matches the naming convention of other parameters.

ChrisNisbet01 committed:
Removed an unused variable.
Moved some code about in preparation for moving the options initialisation out of main().

ChrisNisbet01 committed:
Reduced the scope of a temporary variable.

ChrisNisbet01 committed:
Moved options initialisation out into a separate function.
The manufacturer OUI file is only loaded if the manufacturer command line option is selected.
Manufacturer lookups are now only done if they are going to be displayed.

ChrisNisbet01 committed:
Removed a bit of code duplication relatng to whether or not a node should be printed.

ChrisNisbet01 committed:
Reduced code duplication relating to filtering/logging of APs

ChrisNisbet01 committed:
Removed a superfluous test in the STA printing. The exact same test has already been done just above.

ChrisNisbet01 committed:
Using the mac_address helper function to format and print the station/AP MAC addresses.

ChrisNisbet01 committed:
Oops. A header file was placed in the incorrect directory.

Commit:
Changed some divisions and mods to shifts and bitwise ands.
Replaced references to NB_PRB with the actual array size of the array being referenced.

Commit:
Clean up the initialisation of the AP/STA/NA nodes a little, which was previously split across two functions.
Still need to create a separate intialisation function for each node type.

Commit:
Moved the node initialisation code into separate methods.
Replaced some 'color' magic numbers with a defined value.

Commit:
Grouped AP display related fields into a structure.

Commit:
Grouped filename related fields into a structure.

Commit:
Cleaned up the decloaking code a little.

Commit:
Cleaned up handling of GPS location variables a little. Using an enum instead of magic numbers to index into the location arrays.

Commit:
Formatting changes.

Commit:
Simplfied the battery string code a little.
Untried.

ChrisNisbet01 committed:
A little bit of cleanup in the beacon parser.

ChrisNisbet01 committed:
Did some initial work on adding ubus support. Currently need to continually run a timer that cancels the uboot run.

ChrisNisbet01 committed:
Continuing work on 'ubusifying' the application.
Now using a uloop timer to trigger slow refreshes.

ChrisNisbet01 committed:
Continuing with ubusifying airodump-ng.
Got user input and signals handling done.
Got quit request timeout done.

ChrisNisbet01 committed:
More ubusifying.
Channel hopping and some periodic refreshing now uses ubus timers/fd handling.

ChrisNisbet01 committed:
Active scan an half-second refresh timers are now ubusified.

ChrisNisbet01 committed:
Packet reader has now been ubusified. Using a 10ms timer to read one packet every timeout. Not exactly optimal, but can think more about this later if needs be.

ChrisNisbet01 committed:
Pretty much everything running off ubus callbacks now.
CPU usage is through the roof though.
No events getting sent.
Still need to remove the hack to time out of the blocking uloop_run() call.

ChrisNisbet01 committed:
airodump-ng now 'ubusified'.
Needs cleaning up.
Note that CPU usage is no lower than the original.

ChrisNisbet01 committed:
Now sending events every five seconds that contain AP information.

ChrisNisbet01 committed:
Created a new timer to send a nodes event every 'filter_seconds' seconds.

ChrisNisbet01 committed:
Added stations to the UBUS event.

ChrisNisbet01 committed:
Extracted out the addition of AP and station data to UBUS events.
Started thinking about getting the app to not bother reading user input and/or writing output files when operating in ubus mode. All that needs to happen in UBUS mode is that the AP and STA data is included in perdiodic UBUS events.

ChrisNisbet01 committed:
Moved the main loop into a separate method.

ChrisNisbet01 committed:
Station node UBUS events now include the associated APs BSSID and ESSID at all times.
The decision about what to do with the information is now left to the event listener.

ChrisNisbet01 committed:
Moved the UBUS event code into a separate module.

ChrisNisbet01 committed:
Cleaned up the build a bit. Will build with and without UBUS support now.
Haven't worked out how to do the automake files properly yet.

ChrisNisbet01 committed:
Little bit of cleanup. Now passing 'options' to a couple of functions rather than loads of fields within.

ChrisNisbet01 committed:
Renamed some structure fields. Moved some things around. Reduced the number of args passed to some functions.

ChrisNisbet01 committed:
No longer using a file scoped vairable to store UBUS state.

ChrisNisbet01 committed:
Fixed build issue on OpenWRT.

ChrisNisbet01 committed:
Altered the location that some files are included. These include directives were chucked in so as to get builds under OpenWRT working OK.
Moved them into airodump-ng.h instead.

ChrisNisbet01 committed:
More re-jigging of the includes.
I don't really understand why it all goes wrong if <arpa/inet.h> isn't included, but unless it is I see all sorts of errors like this...
'uint32_t' redeclared as different kind of symbol
 #define cpu_to_be32 __cpu_to_be32

ChrisNisbet01 committed:
Fixed some build warnings.

ChrisNisbet01 committed:
Added UBUS wrapper around some UBUS specific string constant.

ChrisNisbet01 committed:
Removed all UBUS support.

ChrisNisbet01 committed:
Removed all support for the 'wifi_scanner' output file type.

aircrack-ng commented:
After a quick check, here are the initial remarks:
- The OUI file is not mandatory and there is a tool to download it
- There are a lot of unrelated commits, it would be preferable if this had just the related commits, to make it easier to review them
- You removed a lot of "const" from the code, why is that?
- Are the .gitignore changes all necessary?
- Style can be checked easily, we have a .clang-format file, that can be loaded with a lot of code editors or just used in the command line with clang-format. Ideally, with clang-3.9
- Travis noted an error, and it feels like it's just a missing header in one of the Makefiles:

```
/home/travis/build/aircrack-ng/aircrack-ng/dist_build/aircrack-ng-1.5.2_rev-b9ea99c/include/aircrack-ng/support/communications.h:51:42: fatal error: aircrack-ng/osdep/mac_header.h: No such file or directory
```

ChrisNisbet01 committed:
Removed airodump-ng-oui.txt, which should be part of the repo. It can be downloaded as required by the user.

ChrisNisbet01 committed:
Removed some items from gitignore. These files aren't part of the mainline repo.

ChrisNisbet01 commented:
I have removed airodump-ng-oui.txt from the repo and removed some of those changes to .gitignore.
I think you'll find I have _added_ a lot of consts to the code. You would need to point me to the consts you think I have removed.
Yes, there are a lot of commits, but they are all related.
I really don't know enough about the make system to know if I was making the appropriate changes to get all the required headers into the tarball, which is where I think things went wrong.
If these changes are interesting to anybody, they might like to lend a hand and resolve some of the issues.

ChrisNisbet01 committed:
Ran clang-format over the files in src/airodump-ng to format them as per the project standard.

ChrisNisbet01 committed:
Ran clang-format over the other .[ch] files modified during the recent change to format them as per the project standard.

ChrisNisbet01 committed:
Made a guess at adding the required header files as needed into the Makefile.inc files.
'make dist' now completes successfully on my local machine.

ChrisNisbet01 committed:
Hacked in a couple of changes that will hopefully get airodump-ng building under OSX.
As strlcpy() is already provided present under some conditions, it would be better to only include it as required.

ChrisNisbet01 committed:
Added capture_wds-01.cap into test/Makefile.inc so that the newly added airdecap test will pass. It was failing because the .cap file wasn't part of the dist file.

ChrisNisbet01 committed:
Resolved an unsigned enum warning present when building for some platforms.

ChrisNisbet01 committed:
#include <unistd.h> to prevent build error on some platforms.
